keps/sig-testing/3041-node-conformance-and-features/README.md

```diff
@@ -0,0 +1,782 @@
+but this confusion will not be addressed in this KEP.
+
+One difference proposed in this document is to allow NodeConformance to be
+applied to `[Beta]` Features. Since beta features are enabled out of the box and
+in most deployments all Beta features stay enabled, applying Feature label to
+these tests may be misleading. This difference is intended, as we want to make
+sure Beta features as enabled out of the box are tested on PR validation and
+across different environments. This will give a better signal for GA-ing the
+feature.
+
+Proposed definition of `NodeConformance`:
+
+* `[NodeConformance]`: Node-level tests that validating behavior that doesn't
+  depend on specific Node capabilities being present, hardware, or feature set
+  of a dependency (like a container runtime), must be labeled as `[NodeConformance]`.
+  For the ease of test querying, each node-level test that is not testing alpha feature
+  (marked as `[FeatureGate:Foo][Alpha]`) is supposed to be either `NodeConformance`
+  or `Feature`.
+
+### See also
+
+- https://github.com/kubernetes/kubernetes/issues/59001
+- https://docs.google.com/document/d/1BdNVUGtYO6NDx10x_fueRh_DLT-SVdlPC_SsXjYCHOE/edit?usp=sharing
+- https://github.com/kubernetes/community/blob/32a1c14d04ff78684d78b827ac7c49f70352d509/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests
+- https://groups.google.com/g/kubernetes-sig-testing/c/8NVCt9UoV_Q/m/Vqz3_TidAwAJ?utm_medium=email&utm_source=footer
+
+### User Stories (Optional)
+
+#### Story 1
+
+1. New feature gate is introduced as Alpha.
+1. Tests for this functionality are marked as `[FeatureGate:Foo][Alpha]`.
+1. New test grid tab is added to run these tests while enabling the feature gate explicitly.
+1. Feature gate is promoted to Beta and enabled by default.
+1. Tests for this functionality are marked as `[FeatureGate:Foo][Beta]`.
+1. Depending on whether test targets all environments or specific ones,
+   `NodeConformance` or `Feature:` labels are added. Note, the `Feature:` label
+   may (but not necessarily) match the `FeatureGate:` name.
+1. Test infra runs all these tests as part of Features or NodeConformance runs
+   to ensure that default installation of k8s has all the features working.
+1. Test infra runs all tests with every feature gate disabled, catching potential
+   GA features dependencies on the new functionality.
+1. Feature gate is promoted to GA. Feature gate is locked to the value and
+   cannot be disabled.
+1. Tests for this functionality are dropping the labels `[FeatureGate:Foo][Beta]`,
+   while keeping either `NodeConformance` or `Feature:` label.
+
+#### Story 2
+
+Decision making tree for the test labels:
+
+- Is test only works on Linux? Apply `[LinuxOnly]`
+- Is test validate functionality that is controlled by a Feature Gate? Apply `[FeatureGate:Foo][Alpha|Beta|Deprecated]`
+- Is test validate Core API that is enabled by default, GA, and works on any environment? Apply `[Conformance]`
```

Comment:
`Conformance` test require to go through one additional process https://github.com/kubernetes/community/blob/89d1fe2bd23393f0ee5e55d414f1969129a9e52b/contributors/devel/sig-architecture/conformance-tests.md#conformance-testing-in-kubernetes

Reply:
I'm OK to corrrect this in a follow-up, but we should clarify this so the document isn't misleading.

Reply:
Wrote down the mitigation